apputils/containerized.py

```diff
@@ -18,107 +19,116 @@ def __init__(self):
+                environment variable named ``BAR``, which contains the value
+                ``BAZ``, and arg:env_expanded is ``{'FOO': 'BAR'}``, the
+                environment inside the running container will contain an
+                environment variable named ``FOO`` with the value ``BAZ``.
+                In short, the keys are persisted into the container and the
+                values are replaced with the corresponding environment
+                variables, according to the environment running celery.
+            read_only(bool): Set to ``False`` to allow writeable access to
+                filesystem inside container.
+
+        """
+        container_name = self.generate_random_name()
+        env_vars = self.expand_and_update_env_vars(env_vars, env_expand)
+        print("ContainerLauncher: Launching %s from %s" % (container_name,
+                                                           image))
+        result = self.client.containers.run(image, name=container_name,
+                                            detach=False,
+                                            mem_limit=self.mem_limit,
+                                            environment=env_vars)
+        self.client.containers.get(container_name).remove()
+        return result
+
+    @classmethod
+    def generate_random_name(cls):
+        """Generate a string for naming containers."""
+        retval = str(uuid.uuid1()).replace("-", "")
+        return retval
+
+    @classmethod
+    def expand_and_update_env_vars(cls, env_vars, env_expand):
+        """Return a dict object where values are expanded env vars.
+
+        Args:
+            env_vars(dict): This dict is updated by arg:env_expanded
+            env_expanded(dict): Keys are retained, values are expanded.
+        """
+        for k, v in env_expand.items():
+            env_vars[k] = os.getenv(v, "")
+        return env_vars.copy()
+
     def halo_ec2_footprint_csv(self):
         image = ("docker.io/halotools/ec2-halo-delta:%s"
                  % self.ec2_halo_delta_ver)
-        container_name = "ec2_halo_footprint"
-        environment = {
-            "HALO_API_KEY": self.halo_key,
-            "HALO_API_SECRET_KEY": self.halo_secret,
-            "AWS_ACCESS_KEY_ID": self.aws_key,
-            "AWS_SECRET_ACCESS_KEY": self.aws_secret,
-            "HTTPS_PROXY": self.https_proxy,
-            "OUTPUT_FORMAT": "csv"
-        }
-
+        env_expand = {"HALO_API_KEY": "HALO_API_KEY",
+                      "HALO_API_SECRET_KEY": "HALO_API_SECRET_KEY",
+                      "AWS_ACCESS_KEY_ID": "AWS_ACCESS_KEY_ID",
+                      "AWS_SECRET_ACCESS_KEY": "AWS_SECRET_ACCESS_KEY",
+                      "HTTPS_PROXY": "HTTPS_PROXY"}
+        env_vars = {"OUTPUT_FORMAT": "csv"}
         # Populate optional fields to support multi-account inventory.
         optional_fields = ["AWS_ROLE_NAME", "AWS_ACCOUNT_NUMBERS"]
```

Comment:
consider doing this without the loop logic. can you add optional fields to env vars and have it handled in expand_and_update function? Having a loop logic in here handling fields doesn't follow single responsibility principle.